tests/unit/test_calculate_electricity_cost.py

```diff
@@ -22,7 +22,8 @@ async def test_when_electricity_consumption_is_none_then_no_calculation_is_retur
     latest_date,
     period_from,
     period_to,
-    tariff_code
+    tariff_code,
+    False
```

Comment:
We should either have this as an input variable for the existing unit tests, or new unit tests should be added to cover this new scenario

Reply:
Unless I missed something, all of the unit tests used single rate tariffs, so the new "is_smart_meter" parameter doesn't actually have any affect. I've tweaked the integration tests a little but but I don't think there's a quick way to add a unit test for the new code without first adding support for testing dual rate tariffs. Overall, I don't think the test coverage is any worse than it was before.

Let me know what you think.

Reply:
You're correct about the unit tests. Probably an area that needs expanding in the future.